sky/job/utils.py

```diff
@@ -36,7 +36,9 @@
 
 # Add user hash so that two users don't have the same controller VM on
 # shared-account clouds such as GCP.
-SPOT_CONTROLLER_NAME: str = (
+JOB_CONTROLLER_NAME: str = (
+    f'sky-job-controller-{common_utils.get_user_hash()}')
+LEGACY_JOB_CONTROLLER_NAME: str = (
```

Comment:
It seems adding significant overhead for the controller. Also, many clouds do not support cloning a VM at the moment. It might be fine to just ask the user to delete the old controller themselves.